test/minitest/test_minitest_hell.rb

```diff
@@ -0,0 +1,88 @@
+require 'minitest/metametameta'
+require 'minitest/hell'
+
+class TestMinitestHell < MetaMetaMetaTestCase
+  def self.test_order
+    :random
+  end
```

Comment:
The test is basically a copy of the parallelize me test from https://github.com/seattlerb/minitest/blob/master/test/minitest/test_minitest_test.rb#L544. Since that was in a test case that could not be parallelized i copied that property as well just in case more tests get added.

Frankly i do not quite understand how this test works and was mostly trying to come up with something working from what existed before.